src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -390,5 +405,60 @@ private async Task ProcessSearchResultsAsync(IEnumerable<SearchResultEntry> sear
                 await UpdateJobRecordAsync(cancellationToken);
             }
         }
+
+        private async Task<SearchResult> GetGroupPatients(string groupId, List<Tuple<string, string>> queryParametersList, CancellationToken cancellationToken)
+        {
+            if (!queryParametersList.Exists((Tuple<string, string> parameter) => parameter.Item1 == KnownQueryParameterNames.Id))
+            {
+                var patientIds = await GetGroupPatientIds(groupId, cancellationToken);
+                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Id, string.Join(',', patientIds)));
+            }
+
+            using (IScoped<ISearchService> searchService = _searchServiceFactory())
+            {
+                return await searchService.Value.SearchAsync(
+                               resourceType: null,
+                               queryParametersList,
+                               cancellationToken);
+            }
+        }
+
+        private async Task<HashSet<string>> GetGroupPatientIds(string groupId, CancellationToken cancellationToken, HashSet<string> groupsAlreadyChecked = null)
+        {
+            // need to send in the _before param so that if this restarted the same version of the group is retrieved
+
+            if (groupsAlreadyChecked == null)
+            {
+                groupsAlreadyChecked = new HashSet<string>();
+            }
+
+            groupsAlreadyChecked.Add(groupId);
```

Comment:
I am wondering whether groups can contain other groups? Looking at the Group.type definition (https://www.hl7.org/fhir/valueset-group-type.html) seems like it can contain only 6 values.

Reply:
From the reference types allowed for the group entities it seems like more is allowed than what the valueset implies. https://www.hl7.org/fhir/group.html

Reply:
Aah. I see what you are referring to. Missed that